components/accelerator/nvidia/error/xid/README.md

```diff
@@ -0,0 +1,172 @@
+# NVIDIA GPU Xid errors
+
+This `accelerator-nvidia-error-xid` components detects the NVIDIA GPU Xid errors (1) by scanning the dmesg and (2) by using the NVIDIA Management Library (NVML) to catch the Xid events.
+
+The dmesg scan is done with the `dmesg` command and the regex match with the rule:
+
+```regex
+NVRM: Xid.*?: (\d+),
+```
+
+For example, with the following dmesg outputs:
+
+```bash
+dmesg --ctime --nopager --buffer-size 163920
+```
+
+> [Fri Aug 30 11:11:22 2024] nvidia-peermem nv_get_p2p_free_callback:127 ERROR detected invalid context, skipping further processing
+> [Fri Aug 30 11:43:09 2024] NVRM: Xid (PCI:0000:cb:00): 13, pid='<unknown>', name=<unknown>, Graphics SM Warp Exception on (GPC 7, TPC 7, SM 0): Illegal Instruction Parameter
+> [Fri Aug 30 11:43:09 2024] NVRM: Xid (PCI:0000:cb:00): 13, pid='<unknown>', name=<unknown>, Graphics SM Global Exception on (GPC 7, TPC 7, SM 0): Multiple Warp Errors
+> [Fri Aug 30 11:43:09 2024] NVRM: Xid (PCI:0000:cb:00): 13, pid='<unknown>', name=<unknown>, Graphics Exception: ESR 0x57c730=0xc04000b 0x57c734=0x24 0x57c728=0x1f81fb60 0x57c72c=0x1174
+> [Fri Aug 30 11:43:14 2024] NVRM: Xid (PCI:0000:cb:00): 109, pid=1797828, name=pt_main_thread, Ch 00000008, errorString CTX SWITCH TIMEOUT, Info 0x58005
+
+The xid error code will be extracted as follows:
```

Comment:
How about putting a link to this struct? https://github.com/leptonai/gpud/blob/d0be7d6716c1227544af16bb5f8b24fca62e3963/components/accelerator/nvidia/query/xid/xid.go#L11-L22